lib/rubygems/ext/cargo_builder.rb

```diff
@@ -4,26 +4,26 @@
 # over the `cargo rustc` command which takes care of building Rust code in a way
 # that Ruby can use.
 class Gem::Ext::CargoBuilder < Gem::Ext::Builder
-  attr_reader :spec
+  attr_reader :spec, :runner
 
-  def initialize(spec)
+  def initialize(spec, runner: nil)
     @spec = spec
+    @runner = runner || self.class.method(:run)
```

Comment:
I think we should not introduce per builder special arguments. For now, I would prefer to remover runner. We can reintroduce it later if needed in base builder class.

Reply:
OK, fair enough. FWIW, The main reason for this is I would like to be able to configure the builder a bit more when cross compiling, and development.

I can probably get away with just inheriting instead of passing in the runner:

```ruby
class DevBuilder < Gem::Ext::CargoBuilder
  def self.run(...)
end
```

Reply:
Actually, would you be OK with something like this given I remove the special arguments from the builder?

```ruby
builder = Gem::Ext::CargoBuilder.new(spec)
builder.profile = :dev
```

Reply:
Not sure what @simi thanks, but it feels ok to me.